### PR TITLE
Output per bootstrap model scores in vmafossexec.

### PR DESCRIPTION
**Pull request by christosbampis**
Changes to the C++ code so that the output of vmafossexec in xml and json format reports the per bootstrap model scores.

### CODE REVIEW

File: wrapper/src/vmaf.cpp [hunk 1]
```diff
@@ -881,22 +892,41 @@ void BootstrapVmafQualityRunner::_postproc_transform_clip(
...
+        bootstrap_ind_str = bootstrap_ind_strstream.str();
+
+        result.set_scores("vmaf_bootstrap_model_" + bootstrap_ind_str, perModelScore);
+        perModelScore = perModelScore.clear();
+        bootstrap_ind_strstream.str("");
+    }
+
 }
 
 static const char VMAFOSS_DOC_VERSION[] = "1.3.7";
```
Comment (li-zhi) [inline wrapper/src/vmaf.cpp]: Need update the doc version

File: wrapper/src/vmaf.h [hunk 2]
```diff
@@ -63,6 +63,10 @@ class StatVector
 public:
     StatVector() {}
     StatVector(std::vector<double> l): l(l) {}
+    StatVector clear()
```
Comment (li-zhi) [inline wrapper/src/vmaf.h]: What does this mean?

Comment (christosbampis) [reply wrapper/src/vmaf.h]: We define a public method to the StatVector class, which allows us to erase all the contents of a created StatVector object. This is used when we re-use the StatVector perModelScore, which fills up with predictions from each bootstrap model and then is re-used for the next model.

File: wrapper/src/vmaf.cpp (hunk 1 above)
Comment (christosbampis) [reply wrapper/src/vmaf.cpp]: done!

File: python/src/vmaf/core/quality_runner.py [hunk 3]
```diff
@@ -701,8 +701,21 @@ def _get_quality_scores(self, asset):
...
+            # NOTE: bootstrap model indices start from 1
+            bootstrap_model_list.append('{}{:04d}'.format(bootstrap_model_prefix, bootstrap_ind + 1))
+
+        # augment the feature set with any bootstrap models
+        self.FEATURES += bootstrap_model_list
+
         feature_scores = [[] for _ in self.FEATURES]
-        for frame in root.findall('frames/frame'):
+
+        for frame_ind, frame in enumerate(root.findall('frames/frame')):
```
Comment (li-zhi) [inline python/src/vmaf/core/quality_runner.py]: why need frame_ind?

File: wrapper/src/vmaf.cpp [hunk 4]
```diff
@@ -823,83 +827,117 @@ std::unique_ptr<LibsvmNusvrTrainTestModel> BootstrapVmafQualityRunner::_load_mod
...
+        }
+        sprintf(char_buffer, "%04d", j + 1);
+        result.set_scores("vmaf_" + std::string(char_buffer), perModelScore);
+        perModelScore.clear();
+    }
+
 }
 
-static const char VMAFOSS_DOC_VERSION[] = "1.3.7";
+static const char VMAFOSS_DOC_VERSION[] = "1.3.10";
```
Comment (li-zhi) [inline wrapper/src/vmaf.cpp]: Actually it should be 1.3.11 now. Could you also update three things: 1) VERSION, 2) CHANGELOG.md, 3) wrapper/libvmaf.pc?

File: wrapper/src/vmaf.cpp [hunk 5]
```diff
@@ -823,83 +827,117 @@ std::unique_ptr<LibsvmNusvrTrainTestModel> BootstrapVmafQualityRunner::_load_mod
...
     _clip_value(model, baggingScore);
     _clip_value(model, ci95LowScore);
     _clip_value(model, ci95HighScore);
     _clip_value(model, scorePlusDelta);
     _clip_value(model, scoreMinusDelta);
+
+    // clip bootstrap model scores
+    int num_models = predictionStruct.vmafMultiModelPrediction.size();
+
+    for (int model_i; model_i < num_models; model_i++)
```
Comment (HolyWu) [inline wrapper/src/vmaf.cpp]: Shouldn't `model_i` be initialized with value 0?

Comment (christosbampis) [reply wrapper/src/vmaf.cpp]: good catch, thanks! I will be updating this together with some other fixes

**Pull request merged**